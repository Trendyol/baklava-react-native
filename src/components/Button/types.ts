export type ButtonVariantVariantTypes = 'primary' | 'secondary' | 'tertiary';

export type ButtonKindVariantTypes =
  | 'default'
  | 'neutral'
  | 'success'
  | 'danger';

export type ButtonSizeVariantTypes = 's' | 'm' | 'l';
