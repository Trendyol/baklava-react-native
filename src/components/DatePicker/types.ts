export type Day = {
  day: number;
  month: number;
  year: number;
  isActive: boolean;
  isPrevMonth: boolean;
  isNextMonth: boolean;
  isCurrentDate: boolean;
  timestamp: number;
  yearWithMonth: number;
};

export interface CalendarData {
  calendar: (string[] | Day[])[];
  year: number;
  month: number;
  nameOfMonth: string;
}

export type CalendarOptions = {
  year?: number;
  month?: number;
  nameOfWeekdays?: string[];
  nameOfMonths?: string[];
  firstDayOfWeek?: 0 | 1;
};
