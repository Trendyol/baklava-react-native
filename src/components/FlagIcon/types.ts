import { flagIconList } from './list';

export type FlagIconNameType = (typeof flagIconList)[number];
