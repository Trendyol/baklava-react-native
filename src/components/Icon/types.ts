import { iconList } from './list';

export type IconNameType = typeof iconList[number];
