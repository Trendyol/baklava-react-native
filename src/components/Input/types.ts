import { EasingFunction, TextInput } from 'react-native';

export type TextInputRefType = TextInput;

export type InputRefType =
  () => React.MutableRefObject<TextInputRefType | null>;

export type AnimatedViewPropsType = {};

export type AnimatedTextPropsType = {
  style?: any;
};

export type CommonAnimatedPropsTypes = {
  duration: number;
  useNativeDriver: boolean;
  easing: EasingFunction;
};
