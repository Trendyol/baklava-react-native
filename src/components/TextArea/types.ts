import { EasingFunction, TextInput } from 'react-native';

export type TextInputRefType = TextInput;

export type InputRefType =
  () => React.MutableRefObject<TextInputRefType | null>;

export type AnimatedViewPropsType = {};

export type AnimatedTextPropsType = {};

export type CommonAnimatedPropsTypes = {
  duration: number;
  useNativeDriver: boolean;
  easing: EasingFunction;
};
