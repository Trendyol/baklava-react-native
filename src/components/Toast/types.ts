import { VariantProps } from '@ergenekonyigit/restyle';
import { Theme } from '../../theme';

export interface BaseToastData {
  /**
   * variant
   * what color will appear on the screen
   * @type {ToastVariant}
   * @default  ToastVariant.DEFAULT
   * @success ToastVariant.SUCCESS
   * @error - ToastVariant.ERROR
   */
  variant?: VariantProps<Theme, 'toastVariants'>['variant'];
  /**
   * message to appear in toast
   *
   * max length is 110
   */
  text?: string;
  /**
   * how long(ms) will it be visible on the screen
   * @default 7000
   */
  duration?: number;
  /**
   * offset from the bottom of the screen (in px)
   *
   */
  bottomOffset?: number;
}

export interface ToastDataWithoutAction extends BaseToastData {
  action?: never;
  actionText?: never;
}

export interface ToastDataWithAction extends BaseToastData {
  action: (close: () => void) => void;
  actionText: string;
}

export type ToastData = ToastDataWithoutAction | ToastDataWithAction;
export type ToastDataWithId = (ToastDataWithoutAction | ToastDataWithAction) & {
  id: string;
};
export type ShowMethodParams = ToastDataWithoutAction | ToastDataWithAction;

export const TostDurations = {
  SHORT: 7000,
  LONG: 10000,
} as const;

export const ToastDurations = {
  SHORT: 7000,
  LONG: 10000,
} as const;

export type DurationTypes = typeof ToastDurations;
export type Duration = DurationTypes[keyof DurationTypes];

export const ToastVariants = {
  DEFAULT: 'default',
  SUCCESS: 'success',
  ERROR: 'error',
  WARNING: 'warning',
} as const;

export type ToastVariantsTypes = typeof ToastVariants;
export type ToastVariant = ToastVariantsTypes[keyof ToastVariantsTypes];
