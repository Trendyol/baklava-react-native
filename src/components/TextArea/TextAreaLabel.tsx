import React from 'react';
import { Animated } from 'react-native';
import Box from '../Box/Box';
import Text from '../Text/Text';
import { AnimatedTextPropsType, AnimatedViewPropsType } from './types';
import { getLabelColor } from './utils';

export const TextAreaLabel = React.memo(
  ({
    label,
    labelFixed,
    errorState,
    animatedViewProps,
    animatedTextProps,
  }: {
    label?: string | null;
    labelFixed?: boolean;
    errorState: boolean;
    animatedViewProps: AnimatedViewPropsType;
    animatedTextProps: AnimatedTextPropsType | any;
    textAreaHeight: number;
  }) => {
    if (!label) {
      return null;
    }

    const contentSecondaryColor = getLabelColor({ errorState });

    const RenderFixedLabel = () => {
      return (
        <Box flexDirection="row" mb="3xs" testID="fixed-label-box">
          <Text
            variant="subtitle04Medium"
            color={contentSecondaryColor}
            testID="fixed-label">
            {label}
          </Text>
        </Box>
      );
    };

    const RenderOutlinedLabel = () => {
      return (
        <>
          <Animated.View {...animatedViewProps} pointerEvents="none">
            <Box flexDirection="row" pl="m" testID="outlined-label-box">
              <Animated.Text
                style={
                  errorState
                    ? [animatedTextProps.style, { color: 'red' }]
                    : animatedTextProps.style
                }>
                {label}
              </Animated.Text>
            </Box>
          </Animated.View>
        </>
      );
    };

    return labelFixed ? <RenderFixedLabel /> : <RenderOutlinedLabel />;
  },
);
