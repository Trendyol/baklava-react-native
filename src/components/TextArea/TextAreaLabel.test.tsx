import React from 'react';
import { render } from '../../test-utils';
import { TextAreaLabel } from './TextAreaLabel';
import { AnimatedViewPropsType, AnimatedTextPropsType } from './types';

describe('TextArea Label', () => {
  let label: string | null;
  let labelFixed: boolean;
  let errorState: boolean;
  let animatedViewProps: AnimatedViewPropsType;
  let animatedTextProps: AnimatedTextPropsType;
  let textAreaHeight: number;

  beforeEach(() => {
    label = 'Label';
    labelFixed = false;
    errorState = false;
    animatedViewProps = {};
    animatedTextProps = {};
    textAreaHeight = 88;
  });

  test('should render null when not has a label', () => {
    // given
    label = null;

    // when
    const { toJSON } = render(
      <TextAreaLabel
        label={label}
        labelFixed={labelFixed}
        errorState={errorState}
        animatedViewProps={animatedViewProps}
        animatedTextProps={animatedTextProps}
        textAreaHeight={textAreaHeight}
      />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render outlined label correctly', () => {
    // when
    const { toJSON } = render(
      <TextAreaLabel
        label={label}
        labelFixed={labelFixed}
        errorState={errorState}
        animatedViewProps={animatedViewProps}
        animatedTextProps={animatedTextProps}
        textAreaHeight={textAreaHeight}
      />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render outlined label with optional correctly', () => {
    // when
    const { toJSON } = render(
      <TextAreaLabel
        label={label}
        labelFixed={labelFixed}
        errorState={errorState}
        animatedViewProps={animatedViewProps}
        animatedTextProps={animatedTextProps}
        textAreaHeight={textAreaHeight}
      />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render fixed label correctly', () => {
    // given
    labelFixed = true;

    // when
    const { toJSON } = render(
      <TextAreaLabel
        label={label}
        labelFixed={labelFixed}
        errorState={errorState}
        animatedViewProps={animatedViewProps}
        animatedTextProps={animatedTextProps}
        textAreaHeight={textAreaHeight}
      />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render fixed label with optional correctly', () => {
    // given
    labelFixed = true;

    // when
    const { toJSON } = render(
      <TextAreaLabel
        label={label}
        labelFixed={labelFixed}
        errorState={errorState}
        animatedViewProps={animatedViewProps}
        animatedTextProps={animatedTextProps}
        textAreaHeight={textAreaHeight}
      />,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });
});
