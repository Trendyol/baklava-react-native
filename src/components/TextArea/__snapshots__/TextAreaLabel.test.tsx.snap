// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Label should render fixed label correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "marginBottom": 4,
      },
    ]
  }
  testID="fixed-label-box"
>
  <Text
    accessibilityLabel="fixed-label"
    accessible={true}
    style={
      Array [
        Object {
          "color": "#6E7787",
          "fontFamily": "Rubik-Medium",
          "fontSize": 12,
          "fontWeight": "500",
          "lineHeight": 14,
        },
      ]
    }
    testID="fixed-label"
  >
    Label
  </Text>
</View>
`;

exports[`TextArea Label should render fixed label with optional correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "marginBottom": 4,
      },
    ]
  }
  testID="fixed-label-box"
>
  <Text
    accessibilityLabel="fixed-label"
    accessible={true}
    style={
      Array [
        Object {
          "color": "#6E7787",
          "fontFamily": "Rubik-Medium",
          "fontSize": 12,
          "fontWeight": "500",
          "lineHeight": 14,
        },
      ]
    }
    testID="fixed-label"
  >
    Label
  </Text>
</View>
`;

exports[`TextArea Label should render null when not has a label 1`] = `null`;

exports[`TextArea Label should render outlined label correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="none"
  style={Object {}}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingLeft": 16,
        },
      ]
    }
    testID="outlined-label-box"
  >
    <Text
      collapsable={false}
      style={Object {}}
    >
      Label
    </Text>
  </View>
</View>
`;

exports[`TextArea Label should render outlined label with optional correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="none"
  style={Object {}}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "paddingLeft": 16,
        },
      ]
    }
    testID="outlined-label-box"
  >
    <Text
      collapsable={false}
      style={Object {}}
    >
      Label
    </Text>
  </View>
</View>
`;
